source/class/qx/util/format/DateFormat.js

```diff
@@ -109,6 +109,11 @@ qx.Class.define("qx.util.format.DateFormat", {
     }
   },
 
+  destruct() {
+    // has to be cleared explicitly as it is pointing to a complex datastructure
+    this.__formatTree = null;
```

Comment:
Although it doesn't hurt to have this here, it's unnecessary. `__formatTree` is simply an array of objects. There is nothing in it that can't be easily garbage-collected by JavaScript's normal memory management. I have no problem with leaving the destructor here, but the comment is inaccurate, as it indicates that this cleanup is required, when it is not.

Reply:
Revert last commit with @oetiker changes? There is no problem with memory deallocation but I think a lot of compiler warnings which comes out from qooxdoo framework is not good thing if someone wants to find out bad weak places of his app code.

Reply:
What compiler warnings? Is there some warning that was being shown, that this fixes?

Reply:
@derrell these ones:
```
004395 qx.util.format.DateFormat: Missing destruct definition for '__formatTree__P_37_6' in qx.util.format.DateFormat[undefined]: [object Object]
```
I described it in attached issue. But please notice, I set dispose level 3.

Reply:
DateFormat is not only what has this destruct warning. At this moment I also noticed the same problem with disposing `qx.ui.table.Table`  there are a lot of warnings about internal members. Probably we should discuss it on Gitter.